go-controller/pkg/libovsdbops/router.go

```diff
@@ -703,6 +703,36 @@ func DeleteLogicalRouterStaticRoutesWithPredicate(nbClient libovsdbclient.Client
 	return m.Delete(opModels...)
 }
 
+// DeleteLogicalRouterStaticRoutesWithPredicateOps looks up logical router static
+// routes from the cache based on a given predicate, and returns the ops to delete
+// them and remove them from the provided logical router
+func DeleteLogicalRouterStaticRoutesWithPredicateOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation, routerName string, p logicalRouterStaticRoutePredicate) ([]libovsdb.Operation, error) {
```

Comment:
how about changing `DeleteLogicalRouterStaticRoutesWithPredicate` to call this new function and then just 
```
	_, err = TransactAndCheck(nbClient, ops)
	return err
```

Reply:
yup I had the exact same thoughts when I was writing it... yeah I could possible refactor the other one to call this instead :D

Reply:
this is done, resolving.